app/src/main/java/com/example/weatherapp/MainActivity.kt

```diff
@@ -60,6 +64,16 @@ class MainActivity : AppCompatActivity() {
 
     override fun onResume() {
         super.onResume()
+        broadcastMapReceiver = object: BroadcastReceiver() {
+            override fun onReceive(context: Context?, intent: Intent) {
+                val imageByteArray = intent.getByteArrayExtra(Constants.MAP_IMAGE_KEY)
+                val imageBitmap = BitmapFactory.decodeByteArray(imageByteArray, 0, imageByteArray!!.size)
```

Comment:
please do not use `imageByteArray!!`
Instead, check if `imageByteArray` is null one line earlier (use  `if(imageByteArray == null)...` If it's null log an error using logcat and stop the flow.

Reply:
done
I also resolved conflicts and optimized imports across all files